tests/test_file.py

```diff
@@ -68,22 +68,39 @@ def _test_file_put(self, replica: Replica, scheme: str, test_bucket: str, upload
         bundle_uuid = str(uuid.uuid4())
         version = datetime_to_version_format(datetime.datetime.utcnow())
 
-        # should be able to do this twice (i.e., same payload, different UUIDs)
-        self.upload_file(source_url, file_uuid, bundle_uuid=bundle_uuid, version=version)
-        self.upload_file(source_url, str(uuid.uuid4()))
+        with self.subTest(f"{replica}: Created returned when uploading a file with a unique payload, and FQID"):
+            self.upload_file(source_url, file_uuid, bundle_uuid=bundle_uuid, version=version)
+
+        with self.subTest(f"{replica}: Created returned when uploading a file with same payload, and different FQID"):
+            self.upload_file(source_url, str(uuid.uuid4()))
+
+        with self.subTest(f"{replica}: OK returned when uploading a file with the same payload, UUID,  version"):
+            self.upload_file(source_url, file_uuid, bundle_uuid=bundle_uuid,
+                             version=version, expected_code=requests.codes.ok)
+
+        with self.subTest(f"{replica}: OK returned when uploading a file with a different payload and same FQID"):
+            self.upload_file(source_url, file_uuid, version=version, expected_code=requests.codes.ok)
```

Comment:
How is this a different payload? (This isn't a change in your PR, but something I just now noticed)